### PR TITLE
feat(metadata): Include account metadata in Prowler assessments

### PR DESCRIPTION
**Pull request by toniblyx**
### Context 

This code provide additional account information to each csv and json output if Prowler can get information from the AWS organizations management account.
```
  "Account Email": 
  "Account Name":
  "Account ARN": 
  "Account Organization": 
  "Account tags": 
```

### Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfagoagas** [comment]
Also, change PR name to comply with conventional commits format, because the squash commit loads PR name as the message.

File: prowler [hunk 1]
```diff
@@ -644,6 +651,20 @@ if is_junit_output_enabled; then
   prepare_junit_output
 fi
 
+# Backing up credentials
+backupInitialAWSCredentials
```
Comment (jfagoagas) [inline prowler]: This is not needed because `get_orgs_account_details_logic` handles this
```suggestion
```

File: prowler [hunk 2]
```diff
@@ -644,6 +651,20 @@ if is_junit_output_enabled; then
...
+INI_ACCOUNT_TO_ASSUME="${ACCOUNT_TO_ASSUME}"
+if [[ -n "${MANAGEMENT_ACCOUNT_ID}" || -n "${ROLE_TO_ASSUME}" ]]; then
+  ACCOUNT_TO_ASSUME="${MANAGEMENT_ACCOUNT_ID}"
+  get_orgs_account_details_logic $MANAGEMENT_ACCOUNT_ID
+fi
+#Setting account to assume to -A field after getting account metadata
+ACCOUNT_TO_ASSUME="${INI_ACCOUNT_TO_ASSUME}"
+
+#Restoring credentials
+restoreInitialAWSCredentials
```
Comment (jfagoagas) [inline prowler]: This is not needed because `get_orgs_account_details_logic` handles this
```suggestion
```

File: prowler [hunk 3]
```diff
@@ -644,6 +651,20 @@ if is_junit_output_enabled; then
   prepare_junit_output
 fi
 
+# Backing up credentials
+backupInitialAWSCredentials
+#Backing up account to assume
+INI_ACCOUNT_TO_ASSUME="${ACCOUNT_TO_ASSUME}"
+if [[ -n "${MANAGEMENT_ACCOUNT_ID}" || -n "${ROLE_TO_ASSUME}" ]]; then
+  ACCOUNT_TO_ASSUME="${MANAGEMENT_ACCOUNT_ID}"
+  get_orgs_account_details_logic $MANAGEMENT_ACCOUNT_ID
```
Comment (jfagoagas) [inline prowler]: `$MANAGEMENT_ACCOUNT_ID` is not used within 

```suggestion
  get_orgs_account_details_logic `get_orgs_account_details_logic`
```

File: prowler [hunk 4]
```diff
@@ -644,6 +651,20 @@ if is_junit_output_enabled; then
   prepare_junit_output
 fi
 
+# Backing up credentials
+backupInitialAWSCredentials
+#Backing up account to assume
+INI_ACCOUNT_TO_ASSUME="${ACCOUNT_TO_ASSUME}"
```
Comment (jfagoagas) [inline prowler]: Chars are free 😄 

```suggestion
INITIAL_ACCOUNT_TO_ASSUME="${ACCOUNT_TO_ASSUME}"
```

File: prowler [hunk 5]
```diff
@@ -644,6 +651,20 @@ if is_junit_output_enabled; then
...
+if [[ -n "${MANAGEMENT_ACCOUNT_ID}" || -n "${ROLE_TO_ASSUME}" ]]; then
+  ACCOUNT_TO_ASSUME="${MANAGEMENT_ACCOUNT_ID}"
+  get_orgs_account_details_logic $MANAGEMENT_ACCOUNT_ID
+fi
+#Setting account to assume to -A field after getting account metadata
+ACCOUNT_TO_ASSUME="${INI_ACCOUNT_TO_ASSUME}"
```
Comment (jfagoagas) [inline prowler]:
```suggestion
ACCOUNT_TO_ASSUME="${INITIAL_ACCOUNT_TO_ASSUME}"
```

File: include/organizations_metadata [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+# Gets account details with a given ACCOUNT_ID.
+# Prowler requires organizations:ListAccounts* and organizations:ListTagsForResource
+# in the management account in order to get that data. SecurityAudit managed policy includes them.
+
+# Account Tags are in json format with comma, however they are converted to Base64 
+# in order to avoid breaking the CSV or JSON. To use them a post-processor is needed.
+
+get_orgs_account_details(){
+    # -O to get info from Orgs management account needs also (-A) and (-R)
+    if [[ $MANAGEMENT_ACCOUNT_ID ]];then
```
Comment (jfagoagas) [inline include/organizations_metadata]: Why we need this conditional?

`$MANAGEMENT_ACCOUNT_ID` is checked before calling this function.

File: include/organizations_metadata [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        if [[ $(echo "$ACCOUNTS_DETAILS" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "$regx: Access Denied trying to list AWS Organization accounts. Prowler requires organizations:List*" "$regx"
+        fi
+    fi
+}
+
+get_orgs_account_details_logic(){
+    backupInitialAWSCredentials
+    
+    if [[ ${MANAGEMENT_ACCOUNT_ID} ]];then
```
Comment (jfagoagas) [inline include/organizations_metadata]: Why we need this conditional?

`$MANAGEMENT_ACCOUNT_ID` is checked before calling this function.

File: include/organizations_metadata [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+            textInfo "$regx: Access Denied trying to list AWS Organization accounts. Prowler requires organizations:List*" "$regx"
+        fi
+    fi
+}
+
+get_orgs_account_details_logic(){
+    backupInitialAWSCredentials
+    
+    if [[ ${MANAGEMENT_ACCOUNT_ID} ]];then 
+        echo " Prowler is getting details from the AWS Organizations Management Account: $1..."
```
Comment (jfagoagas) [inline include/organizations_metadata]:
```suggestion
        echo " Prowler is getting details from the AWS Organizations Management Account: ${MANAGEMENT_ACCOUNT_ID}..."
```

File: include/organizations_metadata [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        get_orgs_account_details ${MANAGEMENT_ACCOUNT_ID} ${ACCOUNT_NUM}
+    fi 
+    restoreInitialAWSCredentials
+    rm -fr "${TEMP_ORG_STS_ASSUMED_FILE}"
+    rm -fr "${TEMP_ORG_STS_ASSUMED_ERROR}"
```
Comment (jfagoagas) [inline include/organizations_metadata]: Is it needed?

File: include/organizations_metadata [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+# Prowler requires organizations:ListAccounts* and organizations:ListTagsForResource
+# in the management account in order to get that data. SecurityAudit managed policy includes them.
+
+# Account Tags are in json format with comma, however they are converted to Base64 
+# in order to avoid breaking the CSV or JSON. To use them a post-processor is needed.
+
+get_orgs_account_details(){
+    # -O to get info from Orgs management account needs also (-A) and (-R)
+    if [[ $MANAGEMENT_ACCOUNT_ID ]];then
+        if [[ -z $ROLE_TO_ASSUME || -z $ACCOUNT_TO_ASSUME ]]; then
```
Comment (jfagoagas) [inline include/organizations_metadata]: `ACCOUNT_TO_ASSUME` or `MANAGEMENT_ACCOUNT_ID` ?

Comment (jfagoagas) [reply include/organizations_metadata]: Also, this conditional is checked before calling this function.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: include/organizations_metadata [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    else 
+        # Prowler gets only ACTIVE accounts details
+        ACCOUNT_DETAILS_EMAIL=$(echo $ACCOUNTS_DETAILS | jq -r --arg ACCOUNT_ID $2 '.Accounts[] | select(.Status == "ACTIVE") | select(.Id == $ACCOUNT_ID) | "\(.Email)"')
+        ACCOUNT_DETAILS_NAME=$(echo $ACCOUNTS_DETAILS | jq -r --arg ACCOUNT_ID $2 '.Accounts[] | select(.Status == "ACTIVE") | select(.Id == $ACCOUNT_ID) | "\(.Name)"')
+        ACCOUNT_DETAILS_ARN=$(echo $ACCOUNTS_DETAILS | jq -r --arg ACCOUNT_ID $2 '.Accounts[] | select(.Status == "ACTIVE") | select(.Id == $ACCOUNT_ID) | "\(.Arn)"')
+        ACCOUNT_DETAILS_ORG=$(echo $ACCOUNTS_DETAILS | jq -r --arg ACCOUNT_ID $2 '.Accounts[] | select(.Status == "ACTIVE") | select(.Id == $ACCOUNT_ID) | "\(.Arn)"'| awk -F/ '{ print $2 }')
+        ACCOUNT_DETAILS_TAGS=$($AWSCLI $PROFILE_OPT --region $REGION organizations list-tags-for-resource --resource-id $1 --output json | jq -c '. | @base64' 2>&1)
+        if [[ $(echo "$ACCOUNTS_DETAILS" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+            textInfo "$regx: Access Denied trying to list AWS Organization accounts. Prowler requires organizations:List*" "$regx"
+        fi
```
Comment (jfagoagas) [inline include/organizations_metadata]: Is this needed? 

I think it is duplicated from line 33.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: include/organizations_metadata [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            echo "$OPTRED ERROR!$OPTNORMAL - Both Account ID (-A) and IAM Role to assume (-R) must be set with -O"
+            exit 1
+        fi
+        assume_role
+    fi 
+    
+    # Below requires organizations:ListTagsForResource
+    ACCOUNTS_DETAILS=$($AWSCLI $PROFILE_OPT --region $REGION organizations list-accounts --output json 2>&1)
+    if [[ $(echo "$ACCOUNTS_DETAILS" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+        textInfo "$regx: Access Denied trying to list AWS Organization accounts. Prowler requires organizations:List*" "$regx"
```
Comment (jfagoagas) [inline include/organizations_metadata]: Where is `$regx` defined? Is it used?

**Comment by jfagoagas** [comment]
Are we forcing to use the same role name for the Management account (to recover Organizations metadata) and the account to be audited?

**Comment by toniblyx** [comment]
yes, that is on purpose to simplify.

File: include/organizations_metadata (hunk 12 above)
Comment (toniblyx) [reply include/organizations_metadata]: not needed, it is a global service so the default region will work

File: include/organizations_metadata (hunk 9 above)
Comment (toniblyx) [reply include/organizations_metadata]: to clean up the temp files

File: include/organizations_metadata (hunk 12 above)
Comment (jfagoagas) [reply include/organizations_metadata]: But `$regx` is empty in this part. can we delete it?

File: include/organizations_metadata (hunk 9 above)
Comment (jfagoagas) [reply include/organizations_metadata]: yes, but this temp files are not needed anymore. I'm going to delete this part because is handled by `assume_role`

**Review by MrCloudSec — DISMISSED** [review]
LGTM, it is a great feature!

**Review by n4ch04 — DISMISSED** [review]

**Review by n4ch04 — DISMISSED** [review]

**Review by jfagoagas — APPROVED** [review]
Finally!!!!!

**Review by MrCloudSec — APPROVED** [review]
Reviewed!

**Pull request merged**